ViewEnvironmentUI/Tests/ViewEnvironmentObservingTests.swift

```diff
@@ -324,6 +324,70 @@ final class ViewEnvironmentObservingTests: XCTestCase {
         container.setNeedsEnvironmentUpdate()
         XCTAssertEqual(observedEnvironments.count, 2)
     }
+
+    // MARK: - Customizations
+
+    func test_customization() throws {
+        let viewController = TestViewEnvironmentObservingViewController()
+
+        // Customizations should be respected.
+        do {
+            var customizationLifetime: ViewEnvironmentCustomizationLifetime? = viewController
+                .addEnvironmentCustomization {
+                    $0.testContext.number = 200
+                }
+
+            XCTAssertEqual(viewController.environment.testContext.number, 200)
+
+            _ = customizationLifetime // Suppress warning about variable never being read
+            customizationLifetime = nil
+
+            // Customization should be removed when lifetime is deallocated
```

Comment:
You might also want to test explicitly calling `remove()`

Reply:
Sure!

- [ ] add explicit `remove()` call test